.github/workflows/unittests-gpu.yml

```diff
@@ -36,18 +36,25 @@ jobs:
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: us-east-1
 
-      - name: Test project on AWS Batch
+      - name: Extract branch name
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+        id: extract_branch
+
+      - name: Test project on AWS Batch(For push)
+        if: startsWith(steps.extract_branch.outputs.branch, 'PR-') != true
+        run: |
+          python ./tools/batch/submit-job.py --region us-east-1 --job-type g4dn.4x --source-ref ${{ steps.extract_branch.outputs.branch }} --work-dir tools/batch --remote https://github.com/dmlc/gluon-nlp --command "/gluon-nlp/test.sh" --wait | tee > script.log
+
+      - name: Test project on AWS Batch(For pull request)
+        if: startsWith(steps.extract_branch.outputs.branch, 'PR-') == true
         run: |
-          python ./tools/batch/submit-job.py --region us-east-1 --job-type g4dn.4x --source-ref ${{ github.ref }} --work-dir tools/batch --remote https://github.com/dmlc/gluon-nlp --command "/gluon-nlp/test.sh" --wait | tee > script.log
+          python ./tools/batch/submit-job.py --region us-east-1 --job-type g4dn.4x --source-ref refs/pull/${github.event.number}/head --work-dir tools/batch --remote https://github.com/dmlc/gluon-nlp --command "/gluon-nlp/test.sh" --wait | tee > script.log
```

Comment:
Should it be better if we change remote to `${{ github.repository }}` @leezu ?

Reply:
`refs/pull/${{ github.event.pull_request.number }}/head` equivalent has been tested since a while on the 0.x branch, so I think it's preferable. There might be some edge cases when considering multiple repositories.

Reply:
Yes, current logic refers to [this](https://github.com/dmlc/gluon-nlp/blob/0d5c61992180f41eab590e74c7b679980f429292/ci/jenkins/Jenkinsfile_py3-master_gpu_doc#L67-L136) on 0.x branch. What does 'code duplication' mean.

Reply:
duplication means that you have separate, but very similar implementations for pull requests and master branch

Reply:
Make sense. But I tested for `${{ github.repository }}`, it always points to dmlc/gluon-nlp no matter what event(pull or pull_request) is. Which means, use `${{ github.repository }}` here is equivalent to use `https://github.com/dmlc/gluon-nlp`

Reply:
If that's the case, we should merge this in first since our CI is currently blocked.

Reply:
An alternative way is that we can use `${{ github.event.pull_request.head.repo.full_name }}` instead of `${{ github.repository }}` for pull_request.